packages/orama/src/components/defaults.ts

```diff
@@ -25,17 +25,35 @@ export async function getDocumentIndexId(doc: Document): Promise<string> {
 
 export async function validateSchema<S extends Schema = Schema>(doc: Document, schema: S): Promise<boolean> {
   for (const [prop, type] of Object.entries(schema)) {
+    const value = doc[prop]
+
+    if (typeof type === 'string' && type.endsWith('[]')) {
```

Comment:
@ShogunPanda let me know if you agree with this; checking if `type.endsWith('[]')` is clever, but maybe we could just use a dictionary lookup? I feel like it would be a bit cleaner as we add more types

Reply:
Yes, you are right. Let's go that route.

Reply:
Done. I have created two functions in default.ts to return if a type is an array and which is the inner type.